src/utils/categoryFilter.js

```diff
@@ -0,0 +1,13 @@
+export const categoryFilter = (products, filters) => {
+  console.log(filters.selectedCategory);
+  if (
+    filters.selectedCategory.length === 0 ||
+    (filters.selectedCategory.length === 1 &&
+      filters.selectedCategory[0] === null)
```

Comment:
Why you have applied this condition  filters.selectedCategory[0] === null ?? Maybe I think you don't need this condition .

Reply:
Thanks, but I am going to rewrite this functionality in next PR as suggested by more peers.